eclipse-ee4j/jaxrs-api
# TCK Migration: Move jaxrs21/ tests from jakartaee-tck

This is to migrate the tests from jakartaee-tck jaxrs/jaxrs21 folder (not including jaxrs/jaxrs21/platform)
Also the change includes correction to some tests in ee/rs/client/syncinvoker/JAXRSClientIT.java

To test the changes :

export JAVA_HOME=<JDK11_HOME>
export M2_HOME=<MAVEN_HOME>
export PATH=$M2_HOME/bin:$JAVA_HOME/bin:$PATH
export PATH=<project.build.directory>/glassfish6/glassfish/bin:$PATH

jaxrs-tck/
mvn clean install

jersey-tck/
mvn clean verify -Parq-glassfish-managed -Dit.test=jakarta.ws.rs.tck.jaxrs21.**

The tests migrated :
jakartaee-tck/src/com/sun/ts/tests/jaxrs/jaxrs21 


failing test (1) : will be tracked and fixed using https://github.com/eclipse-ee4j/jaxrs-api/issues/1020 

- jakarta.ws.rs.tck.jaxrs21.ee.sse.ssebroadcaster.JAXRSClientIT.sseBroadcastTest


Below are tests that needs to be disabled as per the ts.jtx file. 
They will be disabled along with other skipped tests.
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceWithGenericTypeResponseWhileServerWaitTest
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceWithGenericTypeStringWhileServerWaitTest
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceWithStringCallbackWhileServerWaitTest
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceWithStringClassWhileServerWaitTest
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceWithResponseClassWhileServerWaitTest
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceWithCallbackWhileServerWaitTest
- jakarta.ws.rs.tck.jaxrs21.ee.client.executor.async.JAXRSClientIT.traceTest



Signed-off-by: alwin-joseph <alwin.joseph@oracle.com>
